lib/Onnxifi/Base.cpp

```diff
@@ -219,15 +219,25 @@ void Graph::setTraceEvents(onnxTraceEventList *traceEvents,
     return;
   }
 
+  /// Internally we use steady_clock, but our interface is system_clock
+  /// timestamps. Do a simple conversion.
+  auto steadyTS = TraceEvent::now();
+  auto systemTS = std::chrono::duration_cast<std::chrono::microseconds>(
+                      std::chrono::system_clock::now().time_since_epoch())
+                      .count();
+
+  // Timstamps are uint64_t so branch rather than use abs().
+  int64_t offset =
+      steadyTS > systemTS ? -(steadyTS - systemTS) : (systemTS - steadyTS);
```

Comment:
Consider adding a comment here to explain that you need to make sure you always perform a subtraction that won't underflow, and then negate it if needed. It's not immediately obvious.